.github/workflows/pipelines-unlock.yml

```diff
@@ -80,3 +85,82 @@ jobs:
           infra_live_repo: "."
           infra_live_directory: "."
           deploy_branch_name: ${{ steps.gruntwork_context.outputs.deploy_branch_name }}
+
+  unlock_all:
+    name: Unlock all TF State files
+    if: ${{ inputs.unlock_all }}
+    runs-on: ${{ fromJSON(inputs.runner) }}
+    steps:
+      - name: Checkout Pipelines Actions
+        uses: actions/checkout@v4
+        with:
+          path: pipelines-actions
+          repository: gruntwork-io/pipelines-actions
+          ref: ${{ env.PIPELINES_ACTIONS_VERSION }}
+          token: ${{ secrets.PIPELINES_READ_TOKEN }}
+
+      - name: Check out repo code
+        uses: actions/checkout@v4
+        with:
+          path: infra-live-repo
+          fetch-depth: 0
+
+      - name: Bootstrap Workflow
+        id: gruntwork_context
+        uses: ./pipelines-actions/.github/actions/pipelines-bootstrap
+        with:
+          token: ${{ secrets.PIPELINES_READ_TOKEN }}
+          working_directory: ${{ inputs.working_directory }}
+          terragrunt_command: "force-unlock -force ${{ inputs.lock_id }}"
+          branch: 'main'
+
+      - name: Wipe all dynamodb terraform lock tables
+        shell: bash
+        run: |
+          containsElement () {
+            local e match="$1"
+            shift
+            for e; do [[ "$e" == "$match" ]] && return 0; done
+            return 1
+          }
+
+          echo "Wiping all dynamodb terraform lock tables"
+          # Recursively find every terragrunt hcl
+          UNITS=$(find . -name "terragrunt.hcl")
+          COMPLETED_ACCOUNTS=()
+          for unit in "$UNITS";
+          do
+            dir=$(dirname "$unit")
+            cd $dir
+            ACCOUNTID=$(pipelines config | jq -r ".Authentication.AWSOIDC.AccountID")
+
+            if containsElement $ACCOUNTID "${COMPLETED_ACCOUNTS[@]}"; then
+              echo "Skipping $ACCOUNTID as it has already been processed"
+              continue
+            fi
+
+            auth=$(pipelines auth terragrunt-credentials --ci github-actions --cloud aws --wd .)
+            export AWS_ACCESS_KEY_ID=$(echo $auth | jq -r '.awsCredentials.ACCESS_KEY_ID')
+            export AWS_SECRET_ACCESS_KEY=$(echo $auth | jq -r '.awsCredentials.SECRET_ACCESS_KEY')
+            export AWS_SESSION_TOKEN=$(echo $auth | jq -r '.awsCredentials.SESSION_TOKEN')
+            aws dynamodb delete-table --table-name terraform-locks
+            terragrunt init
```

Comment:
If this works, you're good, but you might want to unset those env vars before running this `terragrunt init`. It might try to do a chained role assumption, but I'm not sure.